src/coap_tinydtls.c

```diff
@@ -693,16 +698,10 @@ verify_ecdsa_key(struct dtls_context_t *dtls_context COAP_UNUSED,
     memcpy(p, other_pub_y, key_size);
     p += key_size;
 
-    assert(p <= (buf + sizeof(buf)));
-
-    get_session_addr(dtls_session, &remote_addr);
-    c_session = coap_session_get_by_peer(t_context->coap_context,
-                                         &remote_addr, dtls_session->ifindex);
-    if (!c_session)
-      return -3;
     coap_lock_callback_ret(ret,
                            t_context->setup_data.validate_cn_call_back(COAP_DTLS_RPK_CERT_CN,
                                buf, p-buf, c_session, 0, 1, t_context->setup_data.cn_call_back_arg));
+    coap_free_type(COAP_DTLS_CONTEXT, buf);
```

Comment:
COAP_DTLS_CONTEXT may be of fixed size (in particular LwIP) and so may not be >= DTLS_CE_LENGTH.

I would use COAP_STRING here.